### DISCUSSION TITLE
Floorp profile

### DISCUSSION THREAD

**Discussion by daemonspudguy** [roddhjav/apparmor.d#403] [Q&A]
I am using the Floorp web browser, which is a fork of Firefox. Could I somehow modify the Firefox profile for Floorp?

**Comment by roddhjav** [comment]
Yes. Have a look at the firefox profile, you will find it include a  include [`app/firefox`](https://github.com/roddhjav/apparmor.d/blob/main/apparmor.d/abstractions/app/firefox) abstraction, it holds most of the firefox profile resource and is common for all firefox based browser. 

So, if you copy the firefox profile to a new `floorp` profile, edit the variables defined in the profile preamble and you should be 80% there.

**Reply by daemonspudguy** [reply]
How do I find things like the name used in dbus?
Edit: nevermind, found it in dbus-monitor.

**Comment by daemonspudguy** [comment]
[floorp.txt](https://github.com/user-attachments/files/15980294/floorp.txt)
I made this profile and enabling it causes Floorp to think it's always running when it isn't.
Floorp starts spewing this in the terminal, as well.
`JavaScript error: resource://gre/modules/XULStore.sys.mjs, line 60: Error: Can't find profile directory.
JavaScript error: resource://gre/modules/XULStore.sys.mjs, line 60: Error: Can't find profile directory.
JavaScript error: resource://gre/modules/XULStore.sys.mjs, line 60: Error: Can't find profile directory.
JavaScript error: resource://gre/modules/XULStore.sys.mjs, line 60: Error: Can't find profile directory.
JavaScript error: resource://gre/modules/XULStore.sys.mjs, line 60: Error: Can't find profile directory.
`

**Reply by roddhjav** [reply]
Add the complain flag to the profile, and [check the logs](https://apparmor.pujol.io/report/).

**Reply by daemonspudguy** [reply]
https://pastebin.com/kKypVq5m

**Reply by daemonspudguy** [reply]
@roddhjav any ideas based on the logs?

**Reply by roddhjav** [reply]
Please, check the logs youserlf: `aa-log -f <your log file> floorp` gives:
```
profile floorp {
  owner @{run}/user/@{uid}/iceauth_HnvdXF r,
  owner @{run}/user/@{uid}/psd/daemonspudguy-floorp-ahuozy03.default/.parentlock wk,
  owner @{run}/user/@{uid}/psd/daemonspudguy-floorp-ahuozy03.default/extension-settings.json w,
  owner @{run}/user/@{uid}/psd/daemonspudguy-floorp-ahuozy03.default/extension-settings.json.tmp rw,
  owner @{run}/user/@{uid}/psd/daemonspudguy-floorp-ahuozy03.default/favicons.sqlite rwk,
  owner @{run}/user/@{uid}/psd/daemonspudguy-floorp-ahuozy03.default/favicons.sqlite-wal w,
  owner @{run}/user/@{uid}/psd/daemonspudguy-floorp-ahuozy03.default/lock w,
  owner @{run}/user/@{uid}/psd/daemonspudguy-floorp-ahuozy03.default/places.sqlite rwk,
  owner @{run}/user/@{uid}/psd/daemonspudguy-floorp-ahuozy03.default/places.sqlite-wal rw,
  owner @{run}/user/@{uid}/psd/daemonspudguy-floorp-ahuozy03.default/storage/default/moz-extension+++@{uuid}^userContextId rwk,
  owner @{run}/user/@{uid}/psd/daemonspudguy-floorp-ahuozy03.default/Workspaces/Workspaces.json r,
}
```

As you correctly guest, it needs integration with psd.

**Comment by daemonspudguy** [comment]
Found out. Turns out I had profile-sync-daemon enabled and those two were not getting along.

**Reply by roddhjav** [reply]
Yea, you may also need to personalise `@{MOUNTS}` to fit with the your mountpoint of psd.
